apps/pay/components/ParsePOSPayment/index.tsx

```diff
@@ -56,10 +59,18 @@ const defaultCurrencyMetadata: Currency = {
   __typename: "Currency",
 }
 
-function ParsePayment({ defaultWalletCurrency, walletId, dispatch, state }: Props) {
+function ParsePayment({
+  defaultWalletCurrency,
+  walletId,
+  dispatch,
+  state,
+  username,
+}: Props) {
   const router = useRouter()
-  const { username, amount, sats, unit, memo, currency } = router.query
-  const { display } = parseDisplayCurrency(router.query)
+  const searchParams = useSearchParams()
+  const { amount, sats, unit, memo } = extractSearchParams(searchParams)
+
+  const display = searchParams?.get("display") ?? localStorage.getItem("display") ?? "USD"
```

Comment:
this line is confusing to me. why would we use the `localStorage` if display is in the query url, or use `USD` otherwise?

Reply:
this was part of older code but this was wrapped in a funciton. the reason is if you have used the app selected display currency as EUR it would be saved in browser and then if you you go to POS next time on url something like 

`pay.blink.sv/username` (we don't have params here) then the localstroage will be used to set your previously seected currency.

Reply:
shouldn't it be then:
```
let display
if (searchParams?.get("display")) {
  display = searchParams?.get("display")
} else {
  if (localStorage.getItem("display"))
    display = localStorage.getItem("display")
else {
    display = "USD"
  }
}
```
}
```